monkey/monkey_island/cc/environment/environment_config.py

```diff
@@ -6,56 +6,48 @@
 from typing import Dict, List
 
 import monkey_island.cc.environment.server_config_generator as server_config_generator
-from monkey_island.cc.consts import MONKEY_ISLAND_ABS_PATH
+from monkey_island.cc.consts import DEFAULT_SERVER_CONFIG_PATH
 from monkey_island.cc.environment.user_creds import UserCreds
 from monkey_island.cc.resources.auth.auth_user import User
 from monkey_island.cc.resources.auth.user_store import UserStore
 
-SERVER_CONFIG_FILENAME = "server_config.json"
-
 
 class EnvironmentConfig:
-    def __init__(self,
-                 server_config: str,
-                 deployment: str,
-                 user_creds: UserCreds,
-                 aws=None):
-        self.server_config = server_config
-        self.deployment = deployment
-        self.user_creds = user_creds
-        self.aws = aws
-
-    @staticmethod
-    def get_from_json(config_json: str) -> EnvironmentConfig:
-        data = json.loads(config_json)
-        return EnvironmentConfig.get_from_dict(data)
+    def __init__(self, file_path):
+        self._server_config_path = os.path.expanduser(file_path)
```

Comment:
So now `server_config.json` resides in user's home directory?

Reply:
The location of `server_config.json` is now configurable at runtime. We need to move away from storing configuration and runtime artifacts alongside the source code, but testing that on every platform is out of scope for this release and PR.

If `file_path` doesn't contain a "~", then `os.path.expanduser()` is effectively a NOP.